gobblin-core/src/main/java/gobblin/source/extractor/filebased/FileBasedSource.java

```diff
@@ -104,11 +104,13 @@ protected void initLogger(SourceState state) {
     Set<String> prevFsSnapshot = Sets.newHashSet();
 
     // Get list of files seen in the previous run
-    if (!previousWorkunits.isEmpty()
-        && previousWorkunits.get(0).getWorkunit().contains(ConfigurationKeys.SOURCE_FILEBASED_FS_SNAPSHOT)) {
-      prevFsSnapshot =
-          previousWorkunits.get(0).getWorkunit().getPropAsSet(ConfigurationKeys.SOURCE_FILEBASED_FS_SNAPSHOT);
-    }
+    if (!previousWorkunits.isEmpty()) {
+      state.setProp(ConfigurationKeys.PREVIOUS_STATE_ID_KEY, previousWorkunits.get(0)
+	      .getProp(ConfigurationKeys.DATASET_STATE_ID_KEY, ConfigurationKeys.DEFAULT_DATASET_STATE_ID));
+      if (previousWorkunits.get(0).getWorkunit().contains(ConfigurationKeys.SOURCE_FILEBASED_FS_SNAPSHOT)) {
+        previousWorkunits.get(0).getWorkunit().getPropAsList(ConfigurationKeys.SOURCE_FILEBASED_FS_SNAPSHOT);
```

Comment:
Grrr.  This is a pin of a PR to keep on top of all the changes and make sure nothing get screwed up.  I'll fix that.